scripts/lib/FileTreeParser.js

```diff
@@ -0,0 +1,176 @@
+// @ts-check
+const dree = require('dree');
+const fs = require('fs').promises;
+const path = require('path');
+
+/** @typedef {{name: string, searchToken: string}} Author */
+/** @typedef {{name: string, searchToken: string}} License */
+/** @typedef {import('dree').Dree} Dree */
+/** @typedef {{parsedContent?: any}} OptionalParsedJSONContent */
+/** @typedef {{children?: DreeWithParsedContent[]} & OptionalParsedJSONContent & Dree} DreeWithParsedContent */
+/** @typedef {{name: string, children: TagsTreeNode[], allChildrenTags: string[] }} TagsTreeNode */
+
+/**
+ * Remove if necessary the BOM character at the beginning of a JSON file content.
+ * @param {string} content
+ */
+const sanitizeJSONContent = (content) => {
+  if (content[0] === '\uFEFF') return content.slice(1);
+
+  return content;
+};
+
+/**
+ * Create a "file tree" by browsing all the files of the specified folder,
+ * ignoring some specific folder names and only listing files with extensions
+ * that we know are useful.
+ *
+ * @param {string} rootPath
+ * @returns {Promise<Dree>}
+ */
+const readFileTree = async (rootPath) => {
+  return await dree.scanAsync(rootPath, {
+    stat: false,
+    normalize: true,
+    followLinks: true,
+    size: false,
+    hash: false,
+    exclude: [
+      /Default size/,
+      /Spritesheet/,
+      /Tilesheet/,
+      /Vector/,
+      /Unimplemented/i,
+      /TODO/i,
+      /Preview\.png/,
+    ],
```

Comment:
I'm not sure this exclude list is relevant here

Reply:
definitely not